src/app/flightsearch/flightsearch.component.ts

```diff
@@ -0,0 +1,538 @@
+            this.noResultsErrorMsg = "There was an error processing your search. Please try again!"
+          }
+        );
+
+        //Now get city information
+        let servicingAreaURL = environment.servicingAreaEndpoint;
+
+        this.httpService.get(servicingAreaURL).subscribe(
+          (res:any) => 
+          {
+            res.forEach((element:any) => {
+              if(!this.cityData.includes(element.servicingArea)){
+                this.cityData.push(element.servicingArea);
+              }
+            });
+            this.cityData.sort();
+          },
+          (err: HttpErrorResponse) =>
+          {
+            console.log(err);
+            this.viewData = [];
+            this.noResultsErrorMsg = "There was an error processing your search. Please try again!"
+          }
+        )
+      }
+    });
+  }
+
+  processGetResults(res: any){
+    let originToDestination = res["Origin to destination"];
+
+    if (originToDestination.length == 0){
+      this.flightsData = [];
+      this.viewData = [];
+      this.noResultsErrorMsg = "We could not find any flights with your search results. Please try again with different parameters";
+      return;
+    }
+    else
+    { 
+      let processedData = this.standardizeResults(originToDestination);
+      if (processedData.length == 0){
+        this.flightsData = [];
+        this.viewData = [];
+        this.noResultsErrorMsg = "We could not find any flights with your search results. Please try again with different parameters";
+        return;
+      }
+      else{
+
+        this.flightsData = processedData;
+        //Tie it into the viewData list
+        this.viewData = [...this.flightsData];
+        this.sortData();
+      }
+    }
+
+    //So round trip flights
+    if (this.tripRB != "One-Way"){
+      let destinationToOrigin = res["Destination to origin"];
+
+      if(destinationToOrigin.length == 0){
+        this.returnFlightsData = [];
+        this.returnTripErrorMsg = "We could not find any flights that return from your destination. Do you still want to view your results?";
+        return;
+      }
+      else
+      { 
+        let processedData = this.standardizeResults(destinationToOrigin);
+        if (processedData.length == 0){
+          this.returnFlightsData = [];
+          this.returnTripErrorMsg = "We could not find any flights that return from your destination. Do you still want to view your results?";
+          return;
+        }
+        else{
+          //Do not make it viewdata; just need to store it
+          this.returnFlightsData = processedData;
+        }
+      }
+    }
+  }
+
+  standardizeResults(data: Array<any>): Array<any>{
+
+    //Create number formatter
+    let formatter = new Intl.NumberFormat("en-US", {
+      style: "currency",
+      currency: "USD",
+    });
+
+    let getCorrectFlights:any = [];
+    if (this.nonStopRB == "Multihop"){
+      data.forEach((element: any) => {
+        let tempObject:any = {};
```

Comment:
this is a good place to use an interface to remove some of the "magic strings" for referencing object properties:
```
interface FlightData {
  cities: string[],
  origin: string,
  ... etc
}
```